tests/v2/integration/project_user_test.go

```diff
@@ -0,0 +1,109 @@
+package validation
+
+import (
+	"testing"
+	"time"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/namespaces"
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	namespaceName = "testnamespace"
+)
+
+type ProjectUserTestSuite struct {
+	suite.Suite
+	testUser *management.User
+	client   *rancher.Client
+	project  *management.Project
+	session  *session.Session
+}
+
+func (p *ProjectUserTestSuite) TearDownSuite() {
+	p.session.Cleanup()
+}
+
+func (p *ProjectUserTestSuite) SetupSuite() {
+	testSession := session.NewSession(p.T())
+	p.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(p.T(), err)
+
+	p.client = client
+
+	projectConfig := &management.Project{
+		ClusterID: "local",
+		Name:      "TestProject",
+	}
+
+	p.T().Log("Create project")
+	testProject, err := client.Management.Project.Create(projectConfig)
+	require.NoError(p.T(), err)
+
+	p.project = testProject
+
+	enabled := true
+	user := &management.User{
+		Username: "testusername",
+		Password: "passwordpasswordd",
+		Name:     "displayname",
+		Enabled:  &enabled,
+	}
+	p.T().Log("Create user")
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(p.T(), err)
+
+	p.testUser = newUser
+}
+
+func (p *ProjectUserTestSuite) TestCreateNamespaceProjectMember() {
+	subSession := p.session.NewSession()
```

Comment:
I don't see this being used in this function?

Reply:
Since our parent session has reference to the subsession, this will clear up resources at the test case level. Or was this not the intention of the subsession?